e2e/tests/erpnext-openmrs-flows.spec.ts

```diff
@@ -0,0 +1,131 @@
+  await openmrs.endPatientVisit();
+
+   // verify
+  await page.goto(`${ERPNEXT_URL}/app/home`);
+  await erpnext.searchQuotation();
+  await expect(quotationStatus).toHaveText('Open');
+});
+
+test('Revising a synced OpenMRS drug order edits the corresponding ERPNext quotation line.', async ({ page }) => {
+  // setup
+  await openmrs.createDrugOrder();
+  await erpnext.open();
+  await expect(page).toHaveURL(/.*home/);
+  await erpnext.searchQuotation();
+  await page.getByRole('link', { name: `${patientName.firstName + ' ' + patientName.givenName}` }).click();
+  const quantity = await page.locator("div.bold:nth-child(4) div:nth-child(2) div");
+  await expect(quantity).toHaveText('12');
+
+  // replay
+  await page.goto(`${O3_URL}`);
+  await openmrs.searchPatient(`${patientName.firstName + ' ' + patientName.givenName}`);
+  await openmrs.editDrugOrder();
+
+  // verify
+  await page.goto(`${ERPNEXT_URL}/app/home`);
+  await erpnext.searchQuotation();
+  await page.getByRole('link', { name: `${patientName.firstName + ' ' + patientName.givenName}` }).click();
+  await expect(quantity).toHaveText('8');
+});
+
+test('Ordering a drug with a free text medication dosage for an OpenMRS patient creates the corresponding ERPNext customer with a filled quotation.', async ({ page }) => {
+  // replay
+  await openmrs.createDrugOrderWithFreeTextDosage();
+
+  // verify
+  await erpnext.open();
+  await expect(page).toHaveURL(/.*home/);
+  await erpnext.searchCustomer();
+  const customer = await page.locator(".bold a:nth-child(1)");
+  await expect(customer).toContainText(`${patientName.firstName + ' ' + patientName.givenName}`);
+  await page.getByRole('link', { name: `${patientName.firstName + ' ' + patientName.givenName}` }).click();
+  await page.locator('#customer-dashboard_tab-tab').click();
+  await page.getByLabel('Dashboard').getByText('Quotation').click();
+  await erpnext.searchQuotation();
+  await expect(page.getByText('Draft').nth(0)).toBeVisible();
+});
+
+test('Discontinuing a synced OpenMRS drug order for an ERPNext customer with a single quotation line removes the corresponding quotation.', async ({ page }) => {
+  // setup
+  await openmrs.createDrugOrder();
+  await erpnext.open();
+  await expect(page).toHaveURL(/.*home/);
+  await erpnext.searchQuotation();
+  await expect(page.getByText(`${patientName.firstName + ' ' + patientName.givenName}`)).toBeVisible();
+
+  // replay
+  await page.goto(`${O3_URL}`);
+  await openmrs.searchPatient(`${patientName.firstName + ' ' + patientName.givenName}`);
+  await openmrs.discontinueDrugOrder();
+
+  // verify
+  await page.goto(`${ERPNEXT_URL}/app/home`);
+  await erpnext.searchQuotation();
+  await expect(page.getByText(`${patientName.firstName + ' ' + patientName.givenName}`)).not.toBeVisible();
+  await expect(page.getByText('No Quotation found')).toBeVisible();
+});
+
+test.afterEach(async ({ page }) => {
+  await openmrs.deletePatient();
+  await page.close();
```

Comment:
Are you doing cleanup in both systems(OpenMRS and ERPNext)?

Reply:
The cleaning is done only in O3, are you suggest deleting the corresponding synced customer/quotation in ERPNext as well?

Reply:
Why only in O3?

Reply:
> Why only in O3?

Let me apply the cleanup in the ERPNext as well. I'll need to do the same for other existing flows in a separate ticket.

Reply:
Done!